### ISSUE TITLE
exort to python converts magic functions

### ISSUE THREAD

**Issue by Exception4U** [jupyter/notebook#3406]
when we export the notebook to python file magic function is also converted into calls of Jupyter API

this is not required for exporting to pure python script

**Comment by mondeja** [comment]
Can you provide a example? This behaviour ensure that can be used later from an IPython environment.

**Comment by Exception4U** [comment]
If I give` %matplotlib inline` as flag in notebook 
and now export this as python file from menu bar then we get following python conversion

```
# coding: utf-8

# In[1]:


get_ipython().magic(u'matplotlib inline')
```
and when we run this python file we get this error

```
Traceback (most recent call last):
  File "Untitled.py", line 7, in <module>
    get_ipython().magic(u'matplotlib inline')
NameError: name 'get_ipython' is not defined
```

So I want to propose how is better way to handle these ipython flag conversion while converting ipynb to py file

**Comment by mondeja** [comment]
Because `get_ipython` is a function that only exists in IPython environment.

**Comment by Exception4U** [comment]
I totally agree so this should not come when we convert to the python file ... isn't this a bug in iynb to python conversion module?

**Comment by Exception4U** [comment]
this issue is not solved yet

**Comment by takluyver** [comment]
We convert the magic commands to make it valid Python *syntax*. We don't get rid of them entirely, because some of them may affect what the code does. So if you've used magic commands, you have to go through and decide what to do with them: whether you can get rid of them entirely, or whether you need to replace them with more general Python code.

For some magic functions, we might be able to guess what you want, but not all of them, and the ones where we do guess we wouldn't always get right. So as `import this` says:

> In the face of ambiguity, refuse the temptation to guess.

**Comment by pharshalp** [comment]
Here is a trick that I use as a workaround...
I put all the magic commands inside try-except blocks
```python
try:
    get_ipython()
    %matplotlib notebook
except NameError:
    pass
```
that way your notebook when converted to a script would run just fine.
